File: strategy/Go/math.go
package math

import (
	"fmt"

	"github.com/kwongjose/DesignPatterns/strategy/Go/addition"
	"github.com/kwongjose/DesignPatterns/strategy/Go/subtract"
)

type operation interface {
	Calculate(x int, y int) int
}

type calcStrategy struct {
	operation
}

var operationStrategy = calcStrategy{}

func Calculate(x int, y int) int {
	return operationStrategy.Calculate(x, y)
}

func SetStrategy(strategy string) {
	switch strategy {
	case "add":
		operationStrategy.Calculate = addition.Calculate
	case "sub":
		operationStrategy.Calculate = subtract.Calculate
	default:
		fmt.Println("Invalid strategy")
	}
}

File: strategy/C#/math.cs
using System;

interface IOperation {
    int calculate(int x, int y);
}

class CalcStrat {
    private IOperation operation; 

    public int run(int x, int y) {
        return this.operation.calculate(x, y);
    }

    public void setStrategy(String strategy) {
        switch (strategy) {
            case "add":
                this.operation = new Addition();
                break;
            case "sub":
                this.operation = new Subtraction();
                break;
            default: 
                Console.WriteLine("Invalid option");
                break;
        }
    }
}